### PR TITLE
dedupe results on read

### PR DESCRIPTION
**Pull request by blockloop**
for some reason duplicate timeseries are returning form tsdb.Read and causing multiple resultsets
to be returned on remote_read. This fix combines resultsets with identical labels

### CODE REVIEW

File: handler/read.go [hunk 1]
```diff
@@ -113,29 +114,46 @@ func toMatchers(query *prompb.Query) []labels.Matcher {
...
 				Value:     v,
 			})
 		}
 
-		ts = append(ts, res)
 	}
+
+	ts := make([]*prompb.TimeSeries, len(ddp))
+	i := 0
+	for _, t := range ddp {
```
Comment (bsnyder788) [inline handler/read.go]: can you use the index variable the for range creates vs manually  incrementing it?

Comment (blockloop) [reply handler/read.go]: I'm iterating a map so the first item is the key, which I don't need

Comment (Hoenn) [reply handler/read.go]: ddp is a map, so wouldn't that range give key, value?

Comment (bsnyder788) [reply handler/read.go]: ah great point...all LGTM

**Pull request merged**